k2/csrc/ragged.cu

```diff
@@ -46,6 +46,13 @@ struct iterator_traits<::RowSplitsDiff> {
 
 namespace k2 {
 
+RaggedShapeDim RaggedShapeDim::Clone() const {
+  RaggedShapeDim ans;
+  ans.row_splits = row_splits.Clone();
+  ans.row_ids = row_ids.Clone();
```

Comment:
not should if `Array.Clone` checks itself valid or not. It's better if we can just copy row_splits or row_ids when they are valid.

Reply:
Looks like Clone() will crash for non-valid input.  How about do: 
`if (row_ids.IsValid()) ans.row_ids = row_ids.Clone();`
I avoided making the Context() call of Array1 check whether the region is non-NULL, because it's called so often (for speed reasons).  Most calls implicity assume the arg is valid.

Reply:
Agreed.

Reply:
Actually I think we shouldn't give RaggedShape a Clone() function at all, because I can't see where it would be
necessary.  If it's ever necessary to modify a RaggedShape in-place (which might invalidate other RaggedShapes it shared arrays with), then just the members that need to be copied can be cloned.

Reply:
Will remove `RaggedShape::Clone()` and `RaggedShapeDim::Clone()`.